### PR TITLE
Provide a fix for WPA click tracking in Agencies card

### PR DESCRIPTION
**Pull request by robertsreberski**
Fixes #25041

#### Changes proposed in this Pull Request:
* Fix link tracking in Agencies card to be triggered on click, not render

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:
* Start the Jetpack Plugin locally on Docker, or however you go about testing the plugin locally
* Apply any Jetpack Plan to your local wordpress site if you don't already have one active
* Go to the Jetpack Dashboard at `/wp-admin/admin.php?page=jetpack#/dashboard`
* Make sure you are on the `At a Glance` tab and scroll down until you see the new Agency module
image
* Ensure that `jetpack_wpa_click` hasn't been fired in your session yet (e.g. with Tracks Vigilante)
* Click the "Learn more" button
* Validate that the `jetpack_wpa_click` event has been fired with `target: 'learn-more-click'`

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx [hunk 1]
```diff
@@ -45,6 +45,11 @@ const AgenciesCard: FC< Props > = ( {
 		[ path, isUserLinked, isOwner ]
 	);
 
+	const handleClick = useCallback( () => {
+		// 2022-11-18: Updating the event name with `-2` prefix to differentiate from the original event (before fix).
+		trackEvent( 'learn-more-click-2' );
```
Comment (jboland88) [inline projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx]: What is the benefit of changing the name of the event versus filtering events by date (pre/ post fix)?
I don't have a strong preference, the `-2` suffix just feels a bit odd IMO.

File: projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx [hunk 2]
```diff
@@ -81,7 +86,7 @@ const AgenciesCard: FC< Props > = ( {
 					<p className="jp-agencies-card__link-button">
 						<Button>
 							<ExternalLink
-								onClick={ trackEvent( 'learn-more-click' ) }
+								onClick={ handleClick }
```
Comment (jboland88) [inline projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx]: For some reason, I am not seeing the event fire on click. I _think_ it has something to do with the the link opening in a new tab.

I tested with the click handler on the `<Button>` element and that seems to work.

**Review by jboland88 — CHANGES_REQUESTED** [review]
The event is not firing on click for me after the change.

File: projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx (hunk 1 above)
Comment (robertsreberski) [reply projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx]: Hmm, the only benefit is that we'll not get old data by accident. I'm not sure what is the best way to solve that; we can also just note this data accident specifying the date it was fixed in the Tracks attribute description.

So I agree, we can drop postfix here.

File: projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx (hunk 2 above)
Comment (robertsreberski) [reply projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx]: Can you make sure of that? Opening the link to jetpack.com page already generates a lot of other events, so it's easy to lose it among others: https://share.cleanshot.com/nyayGo

You can use the "Filter property" text fields in the Vigilante tool. (event `jetpack_wpa_click`).

Additionally, tracking this event depends on javascript, so make sure that you're not using any ad blocker that can disable tracking on the client (js) side.

Comment (CodeyGuyDylan) [reply projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx]: I can confirm I am seeing the event being fired. I am seeing `learn-more-click` not `learn-more-click-2` though like the testing instructions say. That looks correct looking at the code though

**Review by CodeyGuyDylan — DISMISSED** [review]
All LGTM! I think there is a discrepancy from the event name and the name in the testing instructions, but it's not an error or anything

File: projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx (hunk 2 above)
Comment (robertsreberski) [reply projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx]: Yep! I changed it as for the discussion in the other thread (https://github.com/Automattic/jetpack/pull/27503#discussion_r1030857784). I've updated the testing plan for documentation purposes. thanks!

Comment (coder-karen) [reply projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx]: For some reason I'm unable to see the event being fired as well (filtering for `jetpack_wpa_click` and `learn-more-click` to be sure, and testing with ad blocking completely disabled).

Comment (robertsreberski) [reply projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx]: @coder-karen May I ask you to check in Tracks Live View if the event is not recorded on the backend side (to exclude any potential issues with Tracks Vigilante)? I'd check it myself, but I only see this comment 4 hours later, and Tracks Live View shows results up to 2 hours.

And from the analysis of other use cases of `analytics.tracks.recordEvent`, I don't see how the one presented in PR is different (e.g., `jetpack_product_card_disclaimer_click`).

Comment (coder-karen) [reply projects/plugins/jetpack/_inc/client/components/agencies-card/index.tsx]: Following up here too as I followed up in Slack - just to say from my end that checking Tracks Live View didn't show anything after trying again and waiting around an hour.

**Comment by sdixon194** [comment]
Screen Shot 2022-11-28 at 11 14 31 AM

Hmm not sure if I'm testing this correctly, but I also don't see the tracks event fire using Tracks Vigalante. Tested on a JN site, ad block disabled, can see other events firing, but not `jetpack_wpa_click`

**Comment by sdixon194** [comment]
Just updating to say this works locally for me if I test with Jurassic Tube, just not on a JN site using the beta plugin 🤔

**Comment by anomiex** [comment]
Closing the loop from some Slack discussion: it wasn't working in JN but was on local dev sites due to the Gutenberg plugin being installed on the latter but not the former. With that info, @robertsreberski tracked the problem down to https://github.com/WordPress/gutenberg/issues/45212 still affecting WordPress 6.1.1.

**Comment by robertsreberski** [comment]
I've implemented the workaround for the <ExternalLink /> issue, setting up an event handler on parent <Button /> instead. I confirm the event is now showing up on both local Docker and JN envs.

@sdixon194 is it still possible to include that fix in the next release (if it tests well after the recent commit)?

**Review by jboland88 — APPROVED** [review]
This LGTM and tests as described.
I can now see the event firing properly via the browser tools.

**Review by anomiex — APPROVED** [review]
Ok, go for it.

**Comment by sdixon194** [comment]
Cherry picked to release branch in db75deef5556b0b5a59ccf64f6917fd2d2dfb900

**Pull request merged**